### DISCUSSION TITLE
question about ART index

### DISCUSSION THREAD

**Discussion by phasefei** [hyrise/hyrise#2631] [Q&A]
As reading the codes inside hyrise-master/src/lib/storage/index/adaptive_radix_tree/, I found that there is no path compression, deletion/iteration, concurrent support in ART index? It that true? If I was wrong, please point out and I'm sorry for that.

**Comment by Bouncner** [comment]
Hi phasefei,

let me first say that the ART index hasn't been touched for a while and is a candidate for removal. The reason is that we create the ART index per immutable chunks (this, we don't have to care about concurrency too much). But we lately decided to go another direction (an index covers and entire/parts of a table; see "partial index"). We don't delete from indexes. When too much of a MVCC-maintained chunk is deleted, we merge those chunks and create a new index (see `MVCCDeletePluging`), but that is not automated in any way.

That being said, I am not sure what your question concerning compression is. Do you mean compression of the ART index or creating an ART index on compressed data?
Iteration is supported via `AbstractChunkIndex::cbegin()` etc. Or do you mean something else?
